WordPress/Classes/Services/AccountService.m

```diff
@@ -151,20 +151,28 @@ - (void)requestVerificationEmail:(void (^)(void))success failure:(void (^)(NSErr
 /// @name Account creation
 ///-----------------------
 
-- (WPAccount *)createOrUpdateAccountWithUserDetails:(RemoteUser *)remoteUser authToken:(NSString *)authToken
+- (NSManagedObjectID *)createOrUpdateAccountWithUserDetails:(RemoteUser *)remoteUser authToken:(NSString *)authToken
```

Comment:
I like this pattern – I wonder if we should add a test for each one of these methods to ensure that `! objectId.isTemporary`

Reply:
I think that's a good idea. This function here in particular has been covered by unit tests. 

But I guess it's not straightforward to remember adding unit tests for these kind of functions, considering there is no rule to enforce us to do so.

I had an idea the other day that we might be able to help. We could have a simple `PersistedObjectID` type, something like this:

```swift
struct PersistedObjectID<Model> {
  var objectID: NSManangedObjectID

  init(objectID: NSManagedObjectID) {
    // Here we make sure the object id is not temporary.
    assert(!objectID.isTemporary, "call context.obtainPermanentID first")
    self.objectID = objectID
  }

  func object(in context: NSManangedObjectContext) throws -> Model {
    // ...
  }
}
```

Which means APIs that take `NSManagedObjectID` can become strongly typed. For example:
```diff
- func updateAccount(id: NSManagedObjectID)
+ func updateAccount(id: PersistedObjectID<WPAccount>)
```
The compiler would enforce us passing a `WPAccount` object id, instead of any `NSManagedObjectID`.